main.go

```diff
@@ -16,34 +17,55 @@ type DockerHub struct {
 	Repo  string `json:"repo"`
 }
 
+type DockerHubValues struct {
+	SourceType string `json:"source_type"`
+	SourceName string `json:"source_name"`
+}
+
 var (
 	buildDate string
 )
 
+var re = regexp.MustCompile("(.*?/trigger/)[^/]+")
+
 func main() {
 	fmt.Printf("Drone DockerHub Plugin built at %s\n", buildDate)
 
 	vargs := DockerHub{}
+	build := plugin.Build{}
 
 	plugin.Param("vargs", &vargs)
+	plugin.Param("build", &build)
+
 	if err := plugin.Parse(); err != nil {
 		println(err.Error())
 		os.Exit(1)
 	}
 
 	endpoint := fmt.Sprintf("https://registry.hub.docker.com/u/%s/trigger/%s/", vargs.Repo, vargs.Token)
-	values := url.Values{"build": {"true"}}
+	values := DockerHubValues{SourceType: "Branch", SourceName: build.Branch}
```

Comment:
If drone handles an tag event which you can get from the payload you should also post an tag event to the hub